main.js

```diff
@@ -1,5 +1,27 @@
-import oComments from './src/js/comments';
-import oCommentsCount from './src/js/count';
+import Comments from './src/js/comments';
+import CommentCount from './src/js/count';
+
+/**
+ * Initialise the component.
+ *
+ * @param {(HTMLElement|String)} rootEl - The root element to intialise the component in, or a CSS selector for the root element
+ * @param {Object} [opts={}] - An options object for configuring the component
+ * @returns {(Comments|Array<Comments>)|(CommentCount|Array<CommentCount>)} - Comments or CommentCount instance(s)
+ */
+const init = function (rootEl, opts) {
```

Comment:
@rowanmanning, am I on the right track with moving the init function into this file and creating this one entry point?

Reply:
Ah I see what you're moving towards! Yep the single entry point is correct I think, my only suggestion is that maybe we make this a class so that the export has a similar API to other components. So E.g.

`./src/js/comments` -> exports a class named `Comments` (as you currently do)
`./src/js/comment-count` -> exports a class named `CommentCount` (as you currently do)
`main.js`  ->  exports a class named `CommentsController` (or something, I'm bad at names today)

Then when the `CommentsController` class is instantiated in `constructor`, it'd perform the conditional logic below and save the class instance to a property:

```js
this.isCommentCount = false;
if (rootEl.hasAttribute('data-o-comments-count')) {
    this.isCommentCount = true;
    this.instance = new CommentCount(rootEl, opts);
}
this.instance = new Comments(rootEl, opts);
```

Then `CommentsController.init` could keep the same method signature as all the other components.

Does that make sense? Is it a horrible idea?

Reply:
Yes, I like this idea! And it answers my other concerns. I'm going to try this out 😄

Reply:
Let me know if you need anything else 🙂